src/utils/decorators/tooltip-decorator/tooltip-decorator.js

```diff
@@ -163,198 +122,145 @@ const TooltipDecorator = (ComposedComponent) => {
-          this.setState({ tooltipAlign: 'right' });
-        }
+      if (position === 'right') {
+        this.setState({ tooltipPosition: 'top', tooltipAlign: 'right' });
+      } else {
+        this.setState({ tooltipAlign: 'right' });
       }
     }
 
-    /**
-     * Positions tooltip relative to target
-     *
-     * @method positionTooltip
-     * @param {Object} tooltip
-     * @param {Object} target
-     * @return {Void}
-     */
     positionTooltip = () => {
-      if (this.isVisible()) {
-        const tooltip = this.getTooltip(),
-            target = this.getTarget();
-        if (!tooltip || !target) {
-          // Can't find the tooltip or target so hide
-          if (this.state.isVisible) this.setState({ isVisible: false });
-          return;
-        }
-
-        const alignment = this.state.tooltipAlign || this.props.tooltipAlign || 'center',
-            position = this.props.tooltipPosition || 'top',
-            shifts = this.calculatePosition(tooltip, target);
-
-        switch (position) {
-          case 'top':
-            styleElement(tooltip, 'top', append(shifts.verticalY, 'px'));
-            styleElement(tooltip, 'right', 'auto');
-            styleElement(tooltip, 'bottom', 'auto');
-            styleElement(tooltip, 'left', append(shifts[`vertical${startCase(alignment)}`], 'px'));
-            break;
-
-          case 'bottom':
-            styleElement(tooltip, 'top', append(shifts.verticalBottomY, 'px'));
-            styleElement(tooltip, 'bottom', 'auto');
-            styleElement(tooltip, 'left', append(shifts[`vertical${startCase(alignment)}`], 'px'));
-            break;
-
-          case 'left':
-            styleElement(tooltip, 'top', append(shifts[`side${startCase(alignment)}`], 'px'));
-            styleElement(tooltip, 'bottom', 'auto');
-            styleElement(tooltip, 'left', append(shifts[`${position}Horizontal`], 'px'));
-            break;
-
-          case 'right':
-          default:
-            styleElement(tooltip, 'top', append(shifts[`side${startCase(alignment)}`], 'px'));
-            styleElement(tooltip, 'bottom', 'auto');
-            styleElement(tooltip, 'left', append(shifts[`${position}Horizontal`], 'px'));
-        }
+      if (!this.isVisible()) {
+        return;
+      }
+
+      const tooltip = this.getTooltip(),
+          target = this.getTarget();
+
+      if (!tooltip || !target) {
+        if (this.state.isVisible) this.setState({ isVisible: false });
+        return;
+      }
+
+      const alignment = this.state.tooltipAlign || this.props.tooltipAlign || 'center',
+          position = this.state.tooltipPosition || this.props.tooltipPosition || 'top',
+          shifts = this.calculatePosition(tooltip, target);
+
+      this.realignOffscreenTooltip(shifts, tooltip.offsetWidth);
+
+      styleElement(tooltip, 'bottom', 'auto');
+      styleElement(tooltip, 'right', 'auto');
+
+      switch (position) {
+        case 'top':
```

Comment:
Aren't the `top` and `left` cases redundant and can be removed?

Reply:
they aren't redundant - top uses the same methods as bottom and left uses the same as right. There is no `break;` between them.

Reply:
Yep good point 🤦‍♂, `note to self, don't do reviews before you've had coffee`